### ISSUE TITLE
Cori /Theta hangs with setup-env.sh sourced in bashrc

### ISSUE THREAD

**Issue by mamelara** [spack/spack#2388]
Cori hangs whenver setup-env.sh is sourced inside of my bashrc. 
The command `. $SPACK_ROOT/share/spack/setup-env.sh` happens to work fine whenever I manually input them. 

Here's the output when I source bashrc.ext with set -x placed in the setup-env.sh file.

```
_sp_share_dir=$(cd "$(dirname $_sp_source_file)" && pwd)
cd "$(dirname $_sp_source_file)" && pwd)
cd "$(dirname $_sp_source_file)" && pwd
dirname $_sp_source_file)"
dirname $_sp_source_file)
dirname $_sp_source_file
_sp_prefix=$(cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
cd "$(dirname $(dirname $_sp_share_dir))" && pwd
dirname $(dirname $_sp_share_dir))"
dirname $(dirname $_sp_share_dir))
dirname $(dirname $_sp_share_dir)
dirname $_sp_share_dir)
dirname $_sp_share_dir
_spack_pathadd PATH       "${_sp_prefix%/}/bin"
_pa_oldvalue=${PATH:-}
_sp_sys_type=$(spack-python -c 'print(spack.architecture.sys_type())')
spack-python -c 'print(spack.architecture.sys_type())')
spack-python -c 'print(spack.architecture.sys_type())'
```
After that it hangs.

Whenever I try to run any spack command, it hangs and I Ctrl-C out of it. Here's the stack trace:
```
Traceback (most recent call last):
  File "/global/homes/m/mamelara/myspack/bin/spack", line 99, in <module>
    import spack
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/__init__.py", line 74, in <module>
    import spack.repository
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/repository.py", line 45, in <module>
    import spack.spec
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/spec.py", line 114, in <module>
    import spack.store
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/store.py", line 47, in <module>
    import spack.config
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/config.py", line 208, in <module>
    _platform = spack.architecture.platform().name
  File "/global/u2/m/mamelara/myspack/lib/spack/llnl/util/lang.py", line 187, in __call__
    self.cache[args] = self.func(*args)
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/architecture.py", line 530, in platform
    return platform_cls()
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/platforms/cray.py", line 54, in __init__
    _target = self._default_target_from_env()
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/platforms/cray.py", line 109, in _default_target_from_env
    output=str, error=str)
  File "/global/u2/m/mamelara/myspack/lib/spack/spack/util/executable.py", line 171, in __call__
    out, err = proc.communicate()
  File "/global/common/cori/software/python/2.7-anaconda/lib/python2.7/subprocess.py", line 800, in communicate
    return self._communicate(input)
  File "/global/common/cori/software/python/2.7-anaconda/lib/python2.7/subprocess.py", line 1417, in _communicate
    stdout, stderr = self._communicate_with_poll(input)
  File "/global/common/cori/software/python/2.7-anaconda/lib/python2.7/subprocess.py", line 1471, in _communicate_with_poll
    ready = poller.poll()
KeyboardInterrupt
```
Out of ideas on what it could be. It looks like _default_target_from_env is not responding but it works when I don't have spack's shell support activated! Thanks for the help!

**Comment by pramodk** [comment]
Same thing here on Theta @ alcf : 

```
source $SPACK_ROOT/share/spack/setup-env.sh
^CTraceback (most recent call last):
  File "/home/kumbhar/spack/bin/spack", line 80, in <module>
    import spack.main  # noqa
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/__init__.py", line 84, in <module>
    import spack.repository
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/repository.py", line 44, in <module>
    import spack.spec
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/spec.py", line 113, in <module>
    import spack.compilers as compilers
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/compilers/__init__.py", line 36, in <module>
    import spack.config
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/config.py", line 212, in <module>
    _platform = spack.architecture.platform().name
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/llnl/util/lang.py", line 194, in __call__
    self.cache[args] = self.func(*args)
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/architecture.py", line 499, in platform
    return platform_cls()
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/platforms/cray.py", line 79, in __init__
    _target = self._default_target_from_env()
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/platforms/cray.py", line 134, in _default_target_from_env
    output=str, error=str)
  File "/gpfs/mira-home/kumbhar/spack/lib/spack/spack/util/executable.py", line 177, in __call__
    out, err = proc.communicate()
  File "/usr/lib64/python2.7/subprocess.py", line 799, in communicate
    return self._communicate(input)
  File "/usr/lib64/python2.7/subprocess.py", line 1409, in _communicate
    stdout, stderr = self._communicate_with_poll(input)
  File "/usr/lib64/python2.7/subprocess.py", line 1463, in _communicate_with_poll
    ready = poller.poll()
KeyboardInterrupt
```

**Comment by adamjstewart** [comment]
Is this still an issue? Spack's Cray support and shell support have both been completely overhauled in the last 3 years.

**Comment by alalazo** [comment]
Closed by #17386
